src/features/auth/EmailVerification.js

```diff
@@ -0,0 +1,59 @@
+import React, { useState } from 'react';
+import Alert from 'react-bootstrap/Alert';
+import { sendEmailVerification } from '../../app/firebase-configs';
+import useBasicRequestData from '../../app/useBasicRequestData';
+
+export const useEmailVerification = () => {
+  const { user, updatedUser } = useBasicRequestData();
+  const [linkSent, setLinkSent] = useState(false);
+
+  const hasSocialMediaAuthType = user
+    && user.providerData
+    && user.providerData.some(({ providerId }) => providerId !== 'password');
+
+  const { emailVerified = false } = updatedUser || {};
+
+  const sendVerificationLink = () => {
+    sendEmailVerification();
+    setLinkSent(true);
```

Comment:
you can't know if the e-mail was sent, because you didn't wait for the `sendEmailVerification`.